src/coreclr/scripts/superpmi.py

```diff
@@ -1761,6 +1803,199 @@ async def create_one_artifact(jit_path: str, location: str, flags) -> str:
+                    diff_instructions = int(diff_metrics["Diff executed instructions"])
+
+                    logging.info("Total instructions executed by base: {}".format(base_instructions))
+                    logging.info("Total instructions executed by diff: {}".format(diff_instructions))
+                    delta_instructions = diff_instructions - base_instructions
+                    logging.info("Total instructions executed delta: {} ({:.2%} of base)".format(delta_instructions, delta_instructions / base_instructions))
+                    tp_diffs.append((os.path.basename(mch_file), base_instructions, diff_instructions))
+                else:
+                    logging.warning("No metric files present?")
+
+                if not self.coreclr_args.skip_cleanup:
+                    if os.path.isfile(base_metrics_summary_file):
+                        os.remove(base_metrics_summary_file)
+                        base_metrics_summary_file = None
+
+                    if os.path.isfile(diff_metrics_summary_file):
+                        os.remove(diff_metrics_summary_file)
+                        diff_metrics_summary_file = None
+
+            ################################################################################################ end of for mch_file in self.mch_files
+
+        # Report the overall results summary of the tpdiff run
+
+        logging.info("Throughput diff summary:")
+
+        # Construct an overall Markdown summary file.
+
+        if len(tp_diffs) > 0:
+            overall_md_summary_file = create_unique_file_name(self.coreclr_args.spmi_location, "tpdiff_summary", "md")
+            if not os.path.isdir(self.coreclr_args.spmi_location):
+                os.makedirs(self.coreclr_args.spmi_location)
+            if os.path.isfile(overall_md_summary_file):
+                os.remove(overall_md_summary_file)
+
+            with open(overall_md_summary_file, "w") as write_fh:
+                if base_jit_compiler_version != diff_jit_compiler_version:
+                    write_fh.write("Warning: Different compilers used for base and diff JITs. Results may be misleading.\n")
+                    write_fh.write("Base JIT's compiler: {}\n".format(base_jit_compiler_version))
+                    write_fh.write("Diff JIT's compiler: {}\n".format(diff_jit_compiler_version))
+
+                # We write two tables, an overview one with just percentages and
+                # a detailed one that includes raw instruction counts
+                write_fh.write("|Collection|PDIFF|\n")
+                write_fh.write("|---|---|\n")
+                for mch_file, base_instructions, diff_instructions in tp_diffs:
+                    write_fh.write("|{}|{}{:.2f}%|\n".format(
+                        mch_file,
+                        "+" if diff_instructions > base_instructions else "",
+                        (diff_instructions - base_instructions) / base_instructions * 100))
```

Comment:
I wonder if this table should be a "significant summary" and only include collections where the PDIFF shown (at the `.2f` significant digits) is non-zero. The "detailed" can always include full data. This would make the table smaller. If there are no "significant" data it could say that, e.g., "No significant throughput differences found".

Reply:
Makes sense to me, let me work on that.